io/src/lib.rs

```diff
@@ -354,7 +354,7 @@ mod tests {
     fn read_to_limit_greater_than_total_length() {
         let s = "16-byte-string!!".to_string();
         let mut reader = Cursor::new(&s);
-        let mut buf = vec![];
+        let mut buf = Vec::with_capacity(0);
```

Comment:
This does nothing interesting. If we can't predict the capacity we should use `Vec::new()`